src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+         */
+        return $this->client->request('GET', sprintf('https://%s/projects/%d/files/%d/download', $this->endpoint, $this->projectId, $fileId));
+    }
+
+    private function listStrings(int $fileId, int $limit, int $offset): array
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.strings.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.strings.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('GET', sprintf('https://%s/projects/%d/strings', $this->endpoint, $this->projectId), [
+            'query' => [
+                'fileId' => $fileId,
+                'limit' => $limit,
+                'offset' => $offset,
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            $this->logger->error(sprintf('Unable to list strings for file %d in project %d. Message: "%s".', $fileId, $this->projectId, $response->getContent()));
+
+            return [];
+        }
+
+        return $response->toArray()['data'];
+    }
+
+    private function deleteString(int $stringId): ResponseInterface
+    {
+        /*
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.strings.delete (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.strings.delete (Crowdin Enterprise API)
+         */
+        return $this->client->request('DELETE', sprintf('https://%s/projects/%d/strings/%d', $this->endpoint, $this->projectId, $stringId));
+    }
+
+    private function addStorage(string $domain, string $content): int
+    {
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.storages.post (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.storages.post (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/storages', $this->endpoint), [
+            'headers' => [
+                'Crowdin-API-FileName' => urlencode(sprintf('%s.%s', $domain, 'xlf')),
+                'Content-Type' => 'application/octet-stream',
+            ],
+            'body' => $content,
+        ]);
+
+        if (201 !== $response->getStatusCode()) {
+            throw new ProviderException(sprintf('Unable to add a Storage in Crowdin for domain "%s".', $domain), $response);
+        }
+
+        return $response->toArray()['data']['id'];
+    }
+
+    private function getFileList(): array
+    {
+        $result = [];
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('GET', sprintf('https://%s/projects/%d/files', $this->endpoint, $this->projectId));
```

Comment:
May be I miss something, but I think that you don't have to specify the complete URL in all your requests with `$this->client`, as you set a `base_uri` in the Provider Factory. Here, you could only use `projects/%d/files`